terraform/example/bot.tf.example

```diff
@@ -0,0 +1,35 @@
+# Teleport Machine ID Bot creation example
+
+locals {
+  bot_name = "example"
+}
+
+resource "random_password" "bot_token" {
+  length           = 32
+  special          = false
+}
+
+resource "time_offset" "bot_example_token_expiry" {
+  offset_hours = 1
+}
```

Comment:
This comment is non blocking, I don't want to hold the feature but there's something I don't understand about the user flow,

As far as I understand, the bot only use the token on join and continue to work while the token is expired? So the token should be short-lived, or maybe long-lived but deleted on first join?

In this example, the offset is based on the first apply time. Hence the same terraform code cannot be re-applied after 1 hour and will fail with:

```
╷
│ Error: Time validation error
│
│   with teleport_provision_token.bot_example,
│   on main.tf line 101, in resource "teleport_provision_token" "bot_example":
│  101: resource "teleport_provision_token" "bot_example" {
│
│ Attribute metadata.expires value must be in the future
╵
```

If I understand correctly, the users have to re-generate a token during each terraform execution (given the time between two execution is higher than the token expiry) which is not this example's behaviour. If the token have a different name than the previous execution it also means the bot will have to be re-registered.

In its current state, I think the way tokens and bots interact with the descriptive Terraform approach might harm the end user. Users likely work around by creating long-lived tokens, somewhat defeating the Teleport pitch "get rid of long-lived secrets". Maybe allowing a bot to be defined without a token and doing nothing if token expiry is in the past would help, but this is outside of this PR's scope.

Reply:
Machine ID supports a couple different token types. As you mentioned randomized tokens are problematic (though functional, barely) with Terraform. IAM and (as of v11) GitHub OIDC tokens are reusable and are not secret, so the long expiration time doesn't matter.

The plan when we document this is to strongly recommend only using reusable tokens with Terraform.